### ISSUE TITLE
System.ArgumentOutOfRangeException wenn reloading the image list if not yet filled completely

### ISSUE THREAD

**Issue by GoogleCodeExporter** [oozcitak/imagelistview#92]
```
I get the following exception (German only, sorry). The error occurs when 
refilling the imagelist without waiting for it to be fully populated by the 
background thread (see attachment image, too):


System.ArgumentOutOfRangeException

-----------------

Der Index lag außerhalb des Bereichs. Er muss nicht negativ und kleiner als 
die Auflistung sein.
Parametername: index

-----------------

bei System.ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument 
argument, ExceptionResource resource)
   bei System.ThrowHelper.ThrowArgumentOutOfRangeException()
   bei System.Collections.Generic.List`1.get_Item(Int32 index)
   bei Manina.Windows.Forms.ImageListView.OnThumbnailCachedInternal(Guid guid, Boolean error) in C:\Ablage\imagelistview-read-only\ImageListView.cs:Zeile 1367

-----------------

Basically it says there is an array index access error in the file 
"ImageListView.cs", line 1367.

Maybe you can take a look at this.

(Sorry for my lots of error reports...)

Thanks
Uwe
```

Original issue reported on code.google.com by `uwe.k...@gmail.com` on 8 Sep 2010 at 1:50

Attachments:
- [imagelistview-array-error.png](https://storage.googleapis.com/google-code-attachments/imagelistview/issue-92/comment-0/imagelistview-array-error.png)
